From git-commits-head-owner@vger.kernel.org Mon Oct 16 19:21:26 2006
Return-Path: <git-commits-head-owner@vger.kernel.org>
Delivered-To: unknown
Received: from baikonur.stro.at (213.239.196.228) by nancy with IMAP4-SSL; 16
  Oct 2006 17:21:26 -0000
X-Original-To: janitor@sternwelten.at
Delivered-To: janitor@sternwelten.at
Received: from vger.kernel.org (vger.kernel.org [209.132.176.167])
	by baikonur.stro.at (Postfix) with ESMTP id 944245C02E
	for <janitor@sternwelten.at>; Mon, 16 Oct 2006 17:59:46 +0200 (CEST)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
	id S1422683AbWJPP7i (ORCPT <rfc822;janitor@sternwelten.at>);
	Mon, 16 Oct 2006 11:59:38 -0400
Received: (majordomo@vger.kernel.org) by vger.kernel.org id S1422665AbWJPP7i
	(ORCPT <rfc822;git-commits-head-outgoing>);
	Mon, 16 Oct 2006 11:59:38 -0400
Received: from hera.kernel.org ([140.211.167.34]:57264 "EHLO hera.kernel.org")
	by vger.kernel.org with ESMTP id S1161043AbWJPP73 (ORCPT
	<rfc822;git-commits-head@vger.kernel.org>);
	Mon, 16 Oct 2006 11:59:29 -0400
Received: from hera.kernel.org (IDENT:U2FsdGVkX1+FJDthSiW6JuDE9J78ODubLZgj17DiqQs@localhost [127.0.0.1])
	by hera.kernel.org (8.13.7/8.13.7) with ESMTP id k9GFxS54005250
	(version=TLSv1/SSLv3 cipher=DHE-RSA-AES256-SHA bits=256 verify=NO)
	for <git-commits-head@vger.kernel.org>; Mon, 16 Oct 2006 15:59:28 GMT
Received: (from dwmw2@localhost)
	by hera.kernel.org (8.13.7/8.13.1/Submit) id k9GFxSnJ005249
	for git-commits-head@vger.kernel.org; Mon, 16 Oct 2006 15:59:28 GMT
Date: Mon, 16 Oct 2006 15:59:28 GMT
Message-Id: <200610161559.k9GFxSnJ005249@hera.kernel.org>
From: Linux Kernel Mailing List <linux-kernel@vger.kernel.org>
To: git-commits-head@vger.kernel.org
Subject: [Bluetooth] Check if DLC is still attached to the TTY
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
X-Git-Commit: cb19d9ea2ce2bcbe291d3d48e3501dc4f33ba627
X-Git-Parent: b2cfcd75df77b80d9cc3fa84190a350dfa79eb93
Sender: git-commits-head-owner@vger.kernel.org
Precedence: bulk
X-Mailing-List: git-commits-head@vger.kernel.org
X-Virus-Scanned: by Amavis (ClamAV) at stro.at
Status: RO
Content-Length: 1230

commit cb19d9ea2ce2bcbe291d3d48e3501dc4f33ba627
tree 7cd11f8151f7c8bbf04ca74b1d0faaad3e3fd91c
parent b2cfcd75df77b80d9cc3fa84190a350dfa79eb93
author Marcel Holtmann <marcel@holtmann.org> 1160926270 +0200
committer David S. Miller <davem@sunset.davemloft.net> 1160979273 -0700

[Bluetooth] Check if DLC is still attached to the TTY

If the DLC device is no longer attached to the TTY device, then it
makes no sense to go through with changing the termios settings.

Signed-off-by: Marcel Holtmann <marcel@holtmann.org>

 net/bluetooth/rfcomm/tty.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/net/bluetooth/rfcomm/tty.c b/net/bluetooth/rfcomm/tty.c
index 5833b87..b8e3a5f 100644
--- a/net/bluetooth/rfcomm/tty.c
+++ b/net/bluetooth/rfcomm/tty.c
@@ -765,6 +765,9 @@ static void rfcomm_tty_set_termios(struc
 
 	BT_DBG("tty %p termios %p", tty, old);
 
+	if (!dev)
+		return;
+
 	/* Handle turning off CRTSCTS */
 	if ((old->c_cflag & CRTSCTS) && !(new->c_cflag & CRTSCTS)) 
 		BT_DBG("Turning off CRTSCTS unsupported");
-
To unsubscribe from this list: send the line "unsubscribe git-commits-head" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html

